vertexai/function-calling/functioncalling_basic.go

```diff
@@ -12,13 +12,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// Function calling lets developers create a description of a function in their code, then pass
-// that description to a language model in a request.
-//
-// This function call example involves 3 messages:
-// - ask the model to generate a function call request
-// - call the Open API service (simulated in this example)
-// - ask the model to interpret the function call response
+// Function calling allows to improve the model response with the use of external
+// APIs and functions.
 package functioncalling
 
 // [START generativeaionvertexai_gemini_function_calling]
```

Comment:
The Python [sample](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/generative_ai/function_calling/basic_example.py#L23) with the region tag `generativeaionvertexai_gemini_function_calling` demonstrates a text example and is used in 2 doc sections:

* [Examples of function calling / Text examples](https://cloud.google.com/vertex-ai/generative-ai/docs/multimodal/function-calling#python_5)
* [Examples / Send a function declaration](https://cloud.google.com/vertex-ai/generative-ai/docs/model-reference/function-calling#python_1)

We simply cannot re-use this existing Go sample as is as a "Text example" because this sample is using the `Chat` API 🤷 
And since we're not adding new region tags at the moment, I think the best option here is to update this sample to use text responses. This makes it similar to the corresponding Python sample and matches the doc.